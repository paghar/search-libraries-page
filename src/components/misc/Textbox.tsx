interface IInputProps{  
  id?: string,  
  value?:string | number ,
  type: "text" | "number" | "textArea"| "color" ,
  placeholder?:string,
  onChange?:(e:any) => void,
  name?:string,
}
    
const Textbox = (props:IInputProps) => {
  return (
    <input 
      id ={props?.id}
      className="
        bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg
        focus:ring-pink-500 focus:border-pink-500 block w-full pl-10 p-2.5 
        dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 
        dark:text-white dark:focus:ring-pink-500 dark:focus:border-pink-500
        "                       
      value={props?.value}
      type={props?.type}
      placeholder={props?.placeholder} 
      onChange={props.onChange}
      name={props?.name}  
    />  
  );
};
    
export default Textbox;