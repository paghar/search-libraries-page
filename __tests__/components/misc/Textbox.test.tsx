import React from "react";
import {render, fireEvent} from "@testing-library/react";
import Textbox from "../../../src/components/misc/Textbox";

const placeholder = "Enter text";
const onChange = jest.fn();

describe("Button Component", () => {  
  const {container} = render(
    <Textbox
      type="text"
      placeholder={placeholder}
      onChange={onChange}
    />
  );
  it("Renders Snapshot Buttons", () => {    
    expect(container).toMatchSnapshot();
  });
});

describe("Textbox Component", () => {
  test("renders textbox with provided props", () => {    
    const {getByPlaceholderText} = render(
      <Textbox
        type="text"
        placeholder={placeholder}
        onChange={onChange}
      />
    );
    const textbox = getByPlaceholderText(placeholder);
    expect(textbox).toBeInTheDocument();
    expect(textbox).toHaveAttribute("type", "text");
    fireEvent.change(textbox, {target: {value: "Test input"}});
    expect(onChange).toHaveBeenCalled();
  }); 
});