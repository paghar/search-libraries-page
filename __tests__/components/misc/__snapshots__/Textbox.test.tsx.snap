// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders Snapshot Buttons 1`] = `
<div>
  <input
    class="   bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg   focus:ring-pink-500 focus:border-pink-500 block w-full pl-10 p-2.5    dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400    dark:text-white dark:focus:ring-pink-500 dark:focus:border-pink-500   "
    placeholder="Enter text"
    type="text"
    value=""
  />
</div>
`;
